controllers/release/release_adapter.go

```diff
@@ -210,6 +214,45 @@ func (a *Adapter) getApplicationSnapshot() (*appstudioshared.ApplicationSnapshot
 	return applicationSnapshot, nil
 }
 
+// getEnabledTargetReleaseLink returns the ReleaseLink targeted by ReleaseLink in the Release being processed.
+// Only ReleaseLinks with the auto-release label set to true (or missing the label, which we treat the same as
+// having the label and it being set to true) will be searched for. If a matching ReleaseLink is not found or
+// the List operation fails, an error will be returned.
+func (a *Adapter) getEnabledTargetReleaseLink() (*v1alpha1.ReleaseLink, error) {
+	releaseLink, err := a.getReleaseLink()
+	if err != nil {
+		return nil, err
+	}
+
+	releaseLinks := &v1alpha1.ReleaseLinkList{}
+	labelSelector := labels.NewSelector()
+	labelRequirement, err := labels.NewRequirement("release.appstudio.openshift.io/auto-release", selection.NotIn, []string{"false"})
+	if err != nil {
+		return nil, err
+	}
+	labelSelector = labelSelector.Add(*labelRequirement)
+
+	opts := &client.ListOptions{
+		Namespace:     releaseLink.Spec.Target,
+		FieldSelector: fields.OneTermEqualSelector("spec.target", releaseLink.Namespace),
+		LabelSelector: labelSelector,
+	}
+
+	err = a.client.List(a.context, releaseLinks, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, foundReleaseLink := range releaseLinks.Items {
+		if foundReleaseLink.Spec.Application == releaseLink.Spec.Application {
```

Comment:
It seems that the `integration-service` [is able to list using 2 fields](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/integration/integration_adapter.go#L176). In the past we opted for this code because we couldn't make it work. Should we take the chance to improve this code or do you prefer to do it in a different PR?

Reply:
Can you elaborate? Are you saying incorporate this into the fieldselector and no longer require this loop?

Reply:
Exactly. If it's possible (and it seems it is) the code would be much cleaner.

Reply:
Sounds good, I'll test it out and comment back when I have it working (so no need to look at changes until then)

Reply:
I converted this PR to a draft to better track the in-progress status.